Helper/Api/Products.php

```diff
@@ -1130,19 +1137,29 @@ private function processImages(array $flat_product, Product $target, bool $shoul
                     }
                     $shouldUpdate = $this->setGalleryImage($product_image['big_url'], $target, $mainImage);
                     if ($shouldUpdate) {
-                        // Product repo save to trigger gallery images collection to built properly
+                        /**
+                         * Save product via repository fo any given image
+                         * Otherwise new gallery item will not be available
+                         */
                         $this->productRepository->save($target);
+                        $shouldUpdate = false;
                     }
                     $target = $this->productRepository->get($target->getSku());
+                    /**
+                     * Load product item with latest id
+                     * because framework does not allow to receive gallery item via addImageToMediaGallery() call
```

Comment:
```suggestion
                     * because framework does not allow to receive gallery item via addImageToMediaGallery() call
                     * there is nothing to match against, the file path is changed when saving
```